### PR TITLE
Static type checking readme

### PR DESCRIPTION
**Pull request by koleary94**
This is the high level overview of static type checking in javascript. I'm going to make a separate pr for implementing `flow` in `react`.

### CODE REVIEW

**Comment by koleary94** [comment]
@rmutter @jguffey here is the plain file https://github.com/yeti/Engineering/blob/e85184d9205db56d532ea1548c1eb4419b65f238/JavaScript/StaticTypeChecking/README.md

File: JavaScript/StaticTypeChecking/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+## Conclusion
+
+Flow is a new kid on the block. There are new pull requests being opened constantly which are fixing ailments like the one just mentioned. An application - even when perfectly and specifically typed - can still fail when used with third party libraries. There are typings for popular libraries like React, but there is not a type safety guarantee when your application calls upon methods and apis from libraries without set types. You can create those yourself, but the tradeoff is that it takes time. However, there are immense benefits to making the application structure better defined with more contractual agreements.
```
Comment (jguffey) [inline JavaScript/StaticTypeChecking/README.md]: `There are typings for popular libraries like React` - any examples handy?

File: JavaScript/StaticTypeChecking/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Flow can also be added iteratively to **any** project. This is accomplished by just adding a comment to the top of the file. On top of being flexible, flow improves the quality of the codebase. The identity of a solution is presented plainly in code in a readable manner. This makes it easer for any developer unfamiliar with a part of a codebase to form a mental model of how it works in less time. Refactors will also take less time, since the majority of problems caused by refactors in the first place are type errors.
+
+There numerous types that flow exposes that help better specify data. One of these is the `maybe` type, specified by adding a `?` to a type like `?string`. This is essentially saying that the boolean can either be `null` or a `string`. It also allows for union types `string | boolean` and generics `class GenericClass<T> {...}`. There are many more types that can be found in the docs.
```
Comment (rmutter) [inline JavaScript/StaticTypeChecking/README.md]: `There numerous` - grammar

File: JavaScript/StaticTypeChecking/README.md (hunk 1 above)
Comment (koleary94) [inline JavaScript/StaticTypeChecking/README.md]: `Class` is an example. http://www.saltycrane.com/blog/2016/06/flow-type-cheat-sheet/

**Comment by koleary94** [comment]
@rmutter this is more of a high level overview of introducing typing in an app with component based architecture. If the person decides to go through with it, there will be a separate readme for adding it to react. So to answer your question, both.

File: JavaScript/StaticTypeChecking/README.md (hunk 1 above)
Comment (jguffey) [inline JavaScript/StaticTypeChecking/README.md]: We should include this link in the Flow specific document when you post that.

**Comment by jguffey** [comment]
bump @koleary94 @rmutter

**Comment by jguffey** [comment]
bumpety bump

**Comment by rmutter** [comment]
@jguffey at this point we might just want to clean it up and merge it

**Comment by koleary94** [comment]
@rmutter @jguffey sorry i'll get to it today

**Comment by koleary94** [comment]
@jguffey @rmutter addressed feedback

**Pull request merged**